google-oauth-client-java6/src/main/java/com/google/api/client/extensions/java6/auth/oauth2/AuthorizationCodeInstalledApp.java

```diff
@@ -110,7 +152,15 @@ protected void onAuthorization(AuthorizationCodeRequestUrl authorizationUrl) thr
    * @throws IOException I/O exception
    */
   protected void onAuthorization(AuthorizationCodeRequestUrl authorizationUrl) throws IOException {
-    browse(authorizationUrl.build());
+    String url = authorizationUrl.build();
+    Preconditions.checkNotNull(url);
+    System.out.println("Please open the following address in your browser:");
+    System.out.println("  " + url);
+    try {
+      browser.browse(url);
+    } catch (Exception e) {
```

Comment:
We should be able to remove this try/catch and let an `IOException` thrown by the browser be handled by the caller.

The `DefaultBrowser` already swallows the exception so the current behavior is maintained. A developer may want the browser's exception to be handled by the caller.

Reply:
With the update of the `Browser.browse` signature as suggested in your other comment, this will be possible.